### ISSUE TITLE
Shapes are cumulated in Shiny

### ISSUE THREAD

**Issue by stla** [shwilks/r3js#4]
Hello,

Thank you for this very nice package.

I have a Shiny app rendering a `shape3js`, but each time the reactive mesh changes, the new generated plot is added to the previous one. Am I doing something wrong in the code below?

```r
  output[["surface"]] <- renderR3js({
    mesh <- Mesh() # the reactive mesh
    vertices <- mesh[[1L]]
    r3js(
      shape3js(
        plot3js.window(
          plot3js.new(),
          xlim = range(vertices[, 1L]),
          ylim = range(vertices[, 2L]),
          zlim = range(vertices[, 3L]),
          aspect = c(1, 1, 1)
        ),
        vertices = vertices,
        faces = mesh[[2L]],
        normals = NULL,
        col = "maroon"
      )
    )
  })
```

**Comment by stla** [comment]
Here is the full code:

```r
library(rmarchingcubes)
library(r3js)

f <- function(x, y, z, alpha) {
  A <- cospi(alpha)
  B <- sinpi(alpha)
  B2 <- B*B
  z2 <- z*z
  z4 <- z2*z2
  z4*(B2-2*A-2) + 4*x*y^2*(A*B2-4*B2) + x*z2*(A*B2-B2) + 
    3*z2*(A*B2-2*A+2) + x*(B2-A*B^2) + A*B2 - B2
}

h <- function(ρ, θ, ϕ, alpha){
  x <- ρ * cos(θ) * sin(ϕ)
  y <- ρ * sin(θ) * sin(ϕ)
  z <- ρ * cos(ϕ)
  f(x, y, z, alpha)
}

# make grid
nρ <- 150L; nθ <- 250L; nϕ <- 150L
ρ0 <- seq(0, 3.05, length.out = nρ) # ρ runs from 0 to the desired radius
θ0 <- seq(0, 2*pi, length.out = nθ)
ϕ0 <- seq(0, pi, length.out = nϕ) 
G <- expand.grid(Rho = ρ0, Theta = θ0, Phi = ϕ0)


#---- SHINY APP ----####
library(shiny)

# shiny UI ####
ui <- fluidPage(
  sidebarLayout(
    sidebarPanel(
      sliderInput(
        "alpha", "alpha", min = 2, max = 4, value = 3, step = 0.1
      )
    ),
    mainPanel(
      r3jsOutput("surface", height = "512px")
    )
  )
)

# shinyServer ####
server <- function(input, output, session) {

  Mesh <- reactive({
    # voxel
    alpha <- input[["alpha"]] / 4
    voxel <- with(G, array(h(Rho, Theta, Phi, alpha), dim = c(nρ, nθ, nϕ)))
    # 3d contour
    surf <- contour3d(voxel, level = 0, x = ρ0, y = θ0, z = ϕ0)
    # transform to Cartesian coordinates
    ρθϕ <- surf[["vertices"]]
    ρ <- ρθϕ[, 1L]; θ <- ρθϕ[, 2L]; ϕ <- ρθϕ[, 3L] 
    vertices <- cbind(
      ρ * cos(θ) * sin(ϕ),
      ρ * sin(θ) * sin(ϕ),
      ρ * cos(ϕ)
    )
    triangles <- surf[["triangles"]]
    list(vertices, triangles)
  })
  
  output[["surface"]] <- renderR3js({
    mesh <- Mesh()
    vertices <- mesh[[1L]]
    r3js(
      shape3js(
        plot3js.window(
          plot3js.new(),
          xlim = range(vertices[, 1L]),
          ylim = range(vertices[, 2L]),
          zlim = range(vertices[, 3L]),
          aspect = c(1, 1, 1)
        ),
        vertices = vertices,
        faces = mesh[[2L]],
        normals = NULL,
        col = "maroon"
      )
    )
  })
  
}

# shiny app object ----
shinyApp(ui, server)
```

**Comment by shwilks** [comment]
Glad you like the package and thanks for providing the reproducible code, that really helps!

It looks like the problem is that the `.load()` method isn't clearing the previous plot, when doing individual plots it doesn't make much difference but I can see how this is a problem in such shiny apps where the load method gets called multiple times.

It should be easy to fix and push an update to hopefully correct this behaviour, thanks for reporting it!

**Comment by shwilks** [comment]
I think this issue is now resolved in v0.0.2.9000 and the plot will now clear with each update.  At some point I may try and introduce a more efficient function for updating simply the plot components for example but in your use case it seems to update pretty quick anyway.

**Comment by stla** [comment]
It works, thanks!
